storage/connect/frmsg.h

```diff
@@ -122,199 +124,199 @@
+#define MSG_LRECL_TOO_SMALL "Lrecl trop petit (longueur en-t\xeate = %d)"
+#define MSG_MAKE_EMPTY_FILE "G\xe9n\xe9ration du fichier vide %s: %s"
+#define MSG_MAKING          "G\xe9n\xe9ration"
+#define MSG_MALLOC_ERROR    "Allocation m\xe9moire impossible par %s"
 #define MSG_MAP_VIEW_ERROR  "MapViewOfFile %s erreur rc=%d"
 #define MSG_MAXSIZE_ERROR   "Maxsize incalculable sur table ouverte"
-#define MSG_MEM_ALLOC_ERR   "Erreur d'allocation m�moire, taille %s = %d"
-#define MSG_MEM_ALLOC_ERROR "Erreur d'allocation m�moire"
-#define MSG_MISPLACED_QUOTE "Appostrophe mal plac�e ligne %d"
-#define MSG_MISSING_ARG     "Argument manquant pour l'op�rateur %d"
+#define MSG_MEM_ALLOC_ERR   "Erreur d'allocation m\xe9moire, taille %s = %d"
+#define MSG_MEM_ALLOC_ERROR "Erreur d'allocation m\xe9moire"
+#define MSG_MISPLACED_QUOTE "Appostrophe mal plac\xe9e ligne %d"
+#define MSG_MISSING_ARG     "Argument manquant pour l'op\xe9rateur %d"
 #define MSG_MISSING_FIELD   "Champs %d manquant dans %s ligne %d"
 #define MSG_MISSING_FNAME   "Nom du fichier manquant"
 #define MSG_MISSING_NODE    "Noeud %s manquant dans %s"
 #define MSG_MISSING_ROWNODE "Impossible de trouver le noeud de la ligne %d"
 #define MSG_MIS_TAG_LIST    "Liste des balises colonne manquante"
 #define MSG_MUL_MAKECOL_ERR "Erreur logique dans TABMUL::MakeCol"
 #define MSG_NAME_CONV_ERR   "Erreur de convertion du nom de noeud"
-#define MSG_NEW_DOC_FAILED  "Impossible de cr�er le nouveau document"
-#define MSG_NEW_RETURN_NULL "NULL renvoy� par New dans PlugEvalLike"
+#define MSG_NEW_DOC_FAILED  "Impossible de cr\xe9er le nouveau document"
+#define MSG_NEW_RETURN_NULL "NULL renvoy\xe9 par New dans PlugEvalLike"
 #define MSG_NEXT_FILE_ERROR "Erreur en recherche du fichier suivant. rc=%s"
 #define MSG_NONCONT_EXCEPT  "Exception non-continuable"
-#define MSG_NOP_ZLIB_INDEX  "L'indexage d'une table zlib non optimis�e est impossible"
+#define MSG_NOP_ZLIB_INDEX  "L'indexage d'une table zlib non optimis\xe9e est impossible"
 #define MSG_NOT_A_DBF_FILE  "Le fichier n'a pas le format dBASE dbf "
 #define MSG_NOT_FIXED_LEN   "Fichier %s non fixe, len=%d lrecl=%d"
-#define MSG_NO_0DH_HEAD     "0DH manquant en fin d'en-t�te (dbc=%d)"
-#define MSG_NO_ACTIVE_DB    "Pas de base de donn�es active"
-#define MSG_NO_CHAR_FROM    "Conversion de type %d en caract�res impossible"
+#define MSG_NO_0DH_HEAD     "0DH manquant en fin d'en-t\xeate (dbc=%d)"
+#define MSG_NO_ACTIVE_DB    "Pas de base de donn\xe9es active"
+#define MSG_NO_CHAR_FROM    "Conversion de type %d en caract\xe8res impossible"
 #define MSG_NO_DATE_FMT     "Pas de format date pour le valblock de type %d"
-#define MSG_NO_DEF_FNCCOL   "Colonne fonction par d�faut introuvable"
-#define MSG_NO_DEF_PIVOTCOL "Colonne pivot par d�faut introuvable"
-#define MSG_NO_DIR_INDX_RD  "Pas d'acc�s directe des tables %s"
-#define MSG_NO_FEAT_SUPPORT "%s non support� dans cette version"
+#define MSG_NO_DEF_FNCCOL   "Colonne fonction par d\xe9faut introuvable"
```

Comment:
Great catch. The simplest solution, as far as I can tell, to avoid "making this string something else entirely" is to break the string literal into multiple literals, as follows:

```
$ cat > /tmp/test.c <<EOF
#include <stdio.h>

int main(int argc, char *argv[])
{
    // \x escapes are greedy and will consume >2 hex characters
    // The simple way to "interrupt" this greedy behavior is
    // to split the string literal. The following should print,
    // "Hello world cafe", and should NOT give any warnings
    // about exceeding the range of the char type.
    printf("Hello World\x20" "cafe\n");
}
EOF

$ gcc -Wall /tmp/test.c -o /tmp/test && /tmp/test
Hello World cafe
```

The hex escapes introduced in this PR can be fixed by applying this treatment everywhere.

Reply:
Thanks for the suggestion, the treatment of breaking up the string literals to avoid the greedy consumption of hex digits has been applied to the files.

Reply:
Thanks.

The [`diff` from 81494407 (previous tip of this PR branch) to 400e4e70 (latest version)](https://github.com/MariaDB/server/compare/814944076331544346ac9848fb51171f6a337f6e..400e4e708e7ae9bb0a0b8ed9e4c1470381281a8a#diff-8c85fcc7df389ae7114a73ebc7ac22b7dcfeb9e71172e161dad086536f80c079) appears to include all of the following changes:

1. Replacing `\xYZW` in literal strings with `\xYZ""W` in all of the cases where `W` is a legal hexadecimal character (`[0-9a-zA-Z]`) as requested by @LinuxJedi 
2. Realigning the `break` statements to be at the same column as requested by @LinuxJedi 
3. Rebasing on latest `10.6` branch.